src/main/java/model/Movable.java

```diff
@@ -1,32 +1,123 @@
 package model;
 
+import model.directions.Direction;
+
 public class Movable extends Entity {
-	private final Direction direction;
+	private final int speed;
+	private Direction direction;
+	protected boolean hasToMove = false;
 
-	public Movable(Position center, int majorRadius, int minorRadius, Direction direction) {
+	public Movable(Position center, int majorRadius, int minorRadius, int speed, Direction direction) {
 		super(center, majorRadius, minorRadius);
+		this.speed = speed;
 		this.direction = direction;
 	}
 
-	protected void move(Direction direction) {
+	protected void collision() {
```

Comment:
I stronger recommend work on the collision on another branch..

Reply:
Esto tampoco existe.